### PR TITLE
Create klein.gordon.rst, create Klein_Gordon.py, update pinn_forward.rst

### PR DESCRIPTION
**Pull request by ryankrum**
Creating the problem example for the Klein-Gordon equation

### CODE REVIEW

**Comment by lululxvi** [comment]
@anranjiao Pleach check.

**Comment by anranjiao** [comment]
This is a time-dependent PDE.

**Comment by anranjiao** [comment]
> @anranjiao Pleach check.

It's all good now.

File: docs/demos/pinn_forward.rst [hunk 1]
```diff
@@ -44,6 +44,7 @@ Time-dependent PDEs
    pinn_forward/diffusion.1d.resample
    pinn_forward/burgers.rar
    pinn_forward/allen.cahn
+   pinn_forward/klein.gordon.rst
```
Comment (lululxvi) [inline docs/demos/pinn_forward.rst:47]:
```
pinn_forward/klein.gordon
```

File: examples/pinn_forward/Klein_Gordon.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import deepxde as dde
+import numpy as np
+from deepxde.backend import tf
+import matplotlib.pyplot as plt
+from scipy.interpolate import griddata
```
Comment (lululxvi) [inline examples/pinn_forward/Klein_Gordon.py:5]:
```
import deepxde as dde
import matplotlib.pyplot as plt
import numpy as np
from deepxde.backend import tf
from scipy.interpolate import griddata
```

**Comment by ryankrum** [comment]
Just made the changes. Let me know if you want the function spacing changed in the .py file

File: examples/pinn_forward/Klein_Gordon.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+def func(x):
+    return x[:, 0:1] * np.cos(x[:, 1:2])
+
+
+bc = dde.icbc.DirichletBC(geomtime, func, lambda _, on_boundary: on_boundary)
+ic_1 = dde.icbc.IC(geomtime, func, lambda _, on_initial: on_initial)
+ic_2 = dde.icbc.OperatorBC(
+    geomtime,
+    lambda x, y, _: dde.grad.jacobian(y, x, i=0, j=1),
+    lambda x, _: np.isclose(x[1], 0),
```
Comment (lululxvi) [inline examples/pinn_forward/Klein_Gordon.py:36]: `lambda x, _: np.isclose(x[1], 0)` can be replaced by `lambda _, on_initial: on_initial`

File: docs/demos/pinn_forward/klein.gordon.rst [hunk 4]
```diff
@@ -0,0 +1,170 @@
+Klein-Gordon equation
+================
+
+Problem setup
+--------------
+We will solve a Klein-Gordon equation:
```
Comment (lululxvi) [inline docs/demos/pinn_forward/klein.gordon.rst:6]: Add an empty line after `--------------------`

File: docs/demos/pinn_forward/klein.gordon.rst [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+We also specify the following parameters for the equation:
+
+.. math:: \alpha = -1, \beta = 0, \gamma = 1, k = 2.
+
+The reference solution is :math:`y(x, t) = x\cos(t)`.
+
+Implementation
+--------------
+This description goes through the implementation of a solver for the above described Klein-Gordon equation step-by-step.
```
Comment (lululxvi) [inline docs/demos/pinn_forward/klein.gordon.rst:27]: Add empty line here.

File: docs/demos/pinn_forward/klein.gordon.rst [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+        interpolation="nearest",
+        cmap="viridis",
+        extent=[0, 10, -1, 1],
+        origin="lower",
+        aspect="auto",
+    )
+    plt.show()
+    
+Complete code
+--------------
```
Comment (lululxvi) [inline docs/demos/pinn_forward/klein.gordon.rst:169]: Here.

File: examples/pinn_forward/Klein_Gordon.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
+import deepxde as dde
```
Comment (lululxvi) [inline examples/pinn_forward/Klein_Gordon.py:1]: Add the supported backend in the first line.

**Pull request merged**